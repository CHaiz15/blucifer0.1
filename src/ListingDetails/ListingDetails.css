.listing-container {
    margin: 8vh 0 0 0;
    overflow: hidden;
    max-width: 100%;
    position: absolute;
}

.listing-name {
    margin: 2% 0 2% 2%;
    text-shadow: 0 0 1px #000;
}

.listing-title {
    font-weight: bolder;
    font-size: 1.5em;
    margin: 1% 0 1% 2%;
    text-shadow: 0 0 1px #000;
    border-bottom: 1px solid black;
    width: 94%;
}

.features-list {
    list-style: none;
}

.listing-details {
    margin: 0 0 0 2%;
    font-weight: bolder;
    font-size: 1.25em;
    text-shadow: 0 0 1px #000;
}

.listing-features {
    list-style: none;
    margin: 0 0 0 2%;
    font-weight: bolder;
    font-size: 1.25em;
    text-shadow: 0 0 1px #000;
}

.images-container {
    display: flex;
    flex-direction: row;
    height: 40vh;
}

.listing-image1,
.listing-image2,
.listing-image3 {
    box-shadow: 0 0 2px #000;
    width: 33.33333%;
    object-fit: cover;
}

.favorite-btn {
    background-color: #002244;
    color: #fff;
    font-size: 1em;
    height: 20%;
    width: 30%;
}

.switch {
    position: relative;
    display: inline-block;
    width: 60px;
    height: 34px;
    margin: 2% 0 0 0;
}
  
.switch input {
    opacity: 0;
    width: 0;
    height: 0;
}
  
.slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    transition: .4s;
}

.slider:before {
    position: absolute;
    content: "";
    height: 26px;
    width: 26px;
    left: 4px;
    bottom: 4px;
    background-image: url('../assets/Heart.svg');
    transition: .4s;
}

input:checked + .slider {
    background-color: #FB4F14;
}

input:focus + .slider {
    box-shadow: 0 0 1px #FB4F14;
}

input:checked + .slider:before {
    transform: translateX(26px);
}

.slider.round {
    border-radius: 34px;
}

.slider.round:before {
    border-radius: 50%;
}