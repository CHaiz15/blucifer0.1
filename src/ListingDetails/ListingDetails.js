import React from 'react'

const ListingDetails = ({listingId, listings}) => {
    const selectedListing = listings.find(listing => listingId === listing.listing_id)
    const listingFeatures = selectedListing.details.features.map(feature => <li>{feature}</li>)
    return (
        <div>
            <div>
                <h2>{selectedListing.name}</h2>
                <p>Address: {selectedListing.address.street}, {selectedListing.address.zip}</p>
                <p>{selectedListing.details.beds} bed {selectedListing.details.baths} bath</p>
                <p>{selectedListing.details.cost_per_night} / Night</p>
            </div>
            <ul>    
                {listingFeatures}
            </ul>
        </div>
    )
}

export default ListingDetails;