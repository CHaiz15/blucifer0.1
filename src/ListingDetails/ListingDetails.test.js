import React from 'react';
import ListingDetails from './ListingDetails';
import { shallow } from 'enzyme'

describe('ListingDetails', () => {
  let wrapper;
  const mockAddFavoritedListing = jest.fn()
  const mockRemoveFavoritedListing = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
    <ListingDetails 
        key={3}
        addFavoritedListing={mockAddFavoritedListing}
        removeFavoritedListing={mockRemoveFavoritedListing} 
        selectedListingId={3} 
        listings={[{
          listing_id: 3,
          area_id: 590,
          name: "Hip RiNo Party Spot",
          address: {
              street: "2250 Lawrence St",
              zip: "80205"
          },
          details: {
              beds: 3,
              baths: 2.5,
              cost_per_night: 420,
              features: [
                  "hot tub",
                  "espresso machine"
              ]
    }
  }]} 
        favoriteListings={[]}
    />);
  })
  it('should be an instance of ListingDetails component', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it('should start state with default not favorited', () => {
    expect(wrapper.state('favorited')).toEqual(false)
  })

  it('should check if not favorited and change to favorited', () => {
    wrapper.instance().checkFavorited()
    expect(wrapper.state('favorited')).toEqual(true)
  })

  it('should check if favorited and change to unfavorited', () => {
    wrapper.setState({favorited: true})
    wrapper.instance().checkFavorited()
    expect(wrapper.state('favorited')).toEqual(false)
  })
})