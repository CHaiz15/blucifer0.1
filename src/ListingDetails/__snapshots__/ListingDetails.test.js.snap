// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetails should be an instance of ListingDetails component 1`] = `
<div
  className="listing-container"
>
  <div>
    <h2
      className="listing-title"
    >
      Hip RiNo Party Spot
    </h2>
    <p
      className="listing-name"
    >
       
      2250 Lawrence St
      , 
      80205
    </p>
    <div
      className="images-container"
    >
      <img
        alt="listing-image1"
        className="listing-image1"
        src="/images/3_a.jpg"
      />
      <img
        alt="listing-image2"
        className="listing-image2"
        src="/images/3_b.jpg"
      />
      <img
        alt="listing-image3"
        className="listing-image3"
        src="/images/3_c.jpg"
      />
    </div>
    <div
      className="listing-details"
    />
  </div>
  <h4
    className="listing-title"
  >
    FEATURES
  </h4>
  <div
    className="listing-features"
  >
    <p>
      3
       Bed 
      2.5
       Bath
    </p>
    <p>
      $
      420
       / Night
    </p>
    <ul
      className="features-list"
    >
      <li>
        hot tub
      </li>
      <li>
        espresso machine
      </li>
    </ul>
    <label
      className="switch"
    >
      <input
        checked={false}
        onClick={[Function]}
        readOnly={true}
        type="checkbox"
      />
      <span
        className="slider round"
      />
    </label>
  </div>
</div>
`;
